replica/table.cc

```diff
@@ -1108,16 +1108,21 @@ bool table::all_storage_groups_split() {
     return _sg_manager->all_storage_groups_split();
 }
 
-sstables::compaction_type_options::split tablet_storage_group_manager::split_compaction_options() const noexcept {
-    return {[this](dht::token t) {
+future<sstables::compaction_type_options::split> tablet_storage_group_manager::split_compaction_options() const noexcept {
```

Comment:
Let's refactor this patch and make `split_compaction_options()` async as a technical preparation patch, to reduce the churn in the next patch.

Reply:
Sure.